lib/Doctrine/DBAL/Platforms/DB2Platform.php

```diff
@@ -878,4 +880,46 @@ protected function getReservedKeywordsClass()
     {
         return Keywords\DB2Keywords::class;
     }
+    
+    /**
+     * Prepare ORDER BY string for OVER() if applicable
+     * 
+     * @param  string $query
+     * 
+     * @return array|string[]|mixed[]
+     */
+    private function getOrderByForOver($query)
+    {
+        //preset empty array
+        $orderByArray = [];
+
+        //determine if 'ORDER BY' is part of the query
+        $orderByPosition = strrpos($query, 'ORDER BY');
+
+        // ORDER BY found in query string
+        if (false !== $orderByPosition) {
+            $queryArray = preg_split('/[, ]/', substr($query, 0, $orderByPosition -1));
+            $orderByArray = preg_split('/[, ]/', substr($query, $orderByPosition));
+
+            foreach ($orderByArray as $orderIndex => $orderValue) {
+                switch (strtoupper($orderValue)) {
+                    case 'ORDER':
+                    case 'BY':
+                    case 'ASC':
+                    case 'DESC':
+                        break;
+                    case '':
+                        $orderByArray[$orderIndex] = ',';
```

Comment:
With this approach and the `preg_split()` expression above, if the `ORDER BY` columns have any delimiter than exactly ", ", there's a risk of getting malformed SQL. For example, if there's a leading white space(s) before the comma:
```php
var_dump(preg_split('/[, ]/', 'foo , bar , baz'));

/*
array(7) {
  [0] =>
  string(3) "foo"
  [1] =>
  string(0) ""
  [2] =>
  string(0) ""
  [3] =>
  string(3) "bar"
  [4] =>
  string(0) ""
  [5] =>
  string(0) ""
  [6] =>
  string(3) "baz"
}
*/
```
In this case, the resulting SQL will have more than one comma between the columns.

Reply:
...and another fix I need to bang my head against